pkg/hypervctl/system_settings.go

```diff
@@ -4,204 +4,366 @@
+
+	cli := controller.GenerateAddCommand()
+
+	_, stderr, err := powershell.Execute(cli...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add SCSI controller: %w", NewPSError(stderr))
 	}
 
-	controller.systemSettings = s
-	return controller, nil
+	updatedController, err := controller.Update()
+	if err != nil {
+		return nil, fmt.Errorf("failed to update SCSI controller: %w", err)
+	}
+
+	return updatedController, nil
+}
+
+func (s *SystemSettings) GetVM() (*SystemSettings, error) {
+	return GetVMFromName(s.Name)
 }
 
-func (s *SystemSettings) createSystemResourceInternal(settings interface{}, resourceType string, cb func()) error {
-	var service *wmiext.Service
-	var err error
-	if service, err = NewLocalHyperVService(); err != nil {
-		return err
+func getCLI(opts *SystemSettings) []string {
+	if opts.Name == "" {
+		return []string{}
+	}
+
+	params := []string{}
+
+	// VM Name is required
+	params = append(params, fmt.Sprintf("-Name '%s'", opts.Name))
+
+	// String parameters
+	if opts.NewVMName != "" {
+		params = append(params, fmt.Sprintf("-NewVMName '%s'", opts.NewVMName))
+	}
+	if opts.SmartPagingFilePath != "" {
+		params = append(params, fmt.Sprintf("-SmartPagingFilePath '%s'", opts.SmartPagingFilePath))
+	}
+	if opts.SnapshotFileLocation != "" {
+		params = append(params, fmt.Sprintf("-SnapshotFileLocation '%s'", opts.SnapshotFileLocation))
+	}
+	if opts.Notes != "" {
+		params = append(params, fmt.Sprintf("-Notes '%s'", opts.Notes))
 	}
-	defer service.Close()
 
-	if err = populateDefaults(resourceType, settings); err != nil {
-		return err
+	// Numeric parameters
+	if opts.ProcessorCount > 0 {
+		params = append(params, fmt.Sprintf("-ProcessorCount %d", opts.ProcessorCount))
+	}
+	if opts.AutomaticStartDelay > 0 {
+		params = append(params, fmt.Sprintf("-AutomaticStartDelay %d", opts.AutomaticStartDelay))
+	}
+	if opts.AutomaticCriticalErrorActionTimeout > 0 {
+		params = append(params, fmt.Sprintf("-AutomaticCriticalErrorActionTimeout %d", opts.AutomaticCriticalErrorActionTimeout))
 	}
 
-	if cb != nil {
-		cb()
+	// Memory parameters (in bytes)
+	if opts.MemoryStartupBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryStartupBytes %d", opts.MemoryStartupBytes))
+	}
+	if opts.MemoryMinimumBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryMinimumBytes %d", opts.MemoryMinimumBytes))
+	}
+	if opts.MemoryMaximumBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryMaximumBytes %d", opts.MemoryMaximumBytes))
+	}
+	if opts.HighMemoryMappedIoSpace > 0 {
+		params = append(params, fmt.Sprintf("-HighMemoryMappedIoSpace %d", opts.HighMemoryMappedIoSpace))
+	}
+	if opts.LowMemoryMappedIoSpace > 0 {
+		params = append(params, fmt.Sprintf("-LowMemoryMappedIoSpace %d", opts.LowMemoryMappedIoSpace))
 	}
 
-	resourceStr, err := createResourceSettingGeneric(settings, resourceType)
-	if err != nil {
-		return err
+	// Boolean parameters
+	if opts.DynamicMemory {
+		params = append(params, "-DynamicMemory $true")
```

Comment:
it's a [switch parameter type](https://learn.microsoft.com/en-us/powershell/module/hyper-v/set-vm?view=windowsserver2025-ps#-dynamicmemory) so no need to add $true.
Adding it is enough - example ->  https://learn.microsoft.com/en-us/powershell/module/hyper-v/set-vm?view=windowsserver2025-ps#example-2

```suggestion
		params = append(params, "-DynamicMemory")
```